codegen/impl/src/main/kotlin/ConditionGenerator.kt

```diff
@@ -219,4 +206,73 @@ internal class ConditionGenerator @Inject constructor(
             }
         }
     }
+
+    private inner class BooleanExpressionGenerator(
```

Comment:
maybe ut for this?

Reply:
That's a good idea, yes. However, I'm going to finally do #2 (hopefully in 1.4.0), which is going to enable generated code validation against "golden" files. So, expression generation  is going to be automatically covered. So, I propose to cover how various expressions are generated via that mechanism.